storage/block/block.go

```diff
@@ -42,31 +42,49 @@ var (
 type dbBlock struct {
 	sync.RWMutex
 
-	opts          Options
-	ctx           context.Context
+	// NB(r): we cache this instance's stream context finalizer
+	// function to avoid allocating a reference to the fn
+	// every time a read occurs
+	onStreamDoneFinalizer context.Finalizer
+
+	opts Options
+	ctx  context.Context
+
 	startUnixNano int64
 	segment       ts.Segment
 	length        int
 	checksum      uint32
 
 	accessedUnixNano int64
 
-	mergeTarget DatabaseBlock
-
 	retriever  DatabaseShardBlockRetriever
 	retrieveID ts.ID
 
+	// For use in a list of blocks, notably for the wired list. The
+	// doubly linked list pointers are on the block themselves to avoid
+	// creating a heap allocated struct per block.
+	// NB(r): Beware that the list controls the synchronization of these
+	// fields to avoid unnecessary lock contention.
+	// They should never be read or mutated by the block itself.
+	next *dbBlock
```

Comment:
Hey that definitely would be a step up, thanks for the feedback.  Will use an embedded type to encapsulate the next/prev.